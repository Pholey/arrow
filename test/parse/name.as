// Plain identifier
tuesday;
pancake;
main_30;
__hello__;

// Identifier w/type arguments
pancake<int32>;
pancake<>;
pancake<bool, float32>;
pancake<name<int32>>;
pancake<(name<int32>, bool) -> (), (int32), (bool,)>;
