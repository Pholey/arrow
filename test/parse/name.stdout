{
  "tag": "Module",
  "span": "test/parse/name.as:1-12",
  "statements": [
    {
      "tag": "Identifier",
      "span": "test/parse/name.as:2:1-8",
      "text": "tuesday"
    },
    {
      "tag": "Identifier",
      "span": "test/parse/name.as:3:1-8",
      "text": "pancake"
    },
    {
      "tag": "Identifier",
      "span": "test/parse/name.as:4:1-8",
      "text": "main_30"
    },
    {
      "tag": "Identifier",
      "span": "test/parse/name.as:5:1-10",
      "text": "__hello__"
    },
    {
      "tag": "Identifier",
      "span": "test/parse/name.as:8:1-15",
      "text": "pancake"
    },
    {
      "tag": "Identifier",
      "span": "test/parse/name.as:9:1-10",
      "text": "pancake"
    },
    {
      "tag": "Identifier",
      "span": "test/parse/name.as:10:1-23",
      "text": "pancake"
    },
    {
      "tag": "Identifier",
      "span": "test/parse/name.as:11:1-21",
      "text": "pancake"
    },
    {
      "tag": "Identifier",
      "span": "test/parse/name.as:12:1-53",
      "text": "pancake"
    }
  ]
}
